api/preprints/serializers.py

```diff
@@ -112,8 +112,9 @@ def update(self, preprint, validated_data):
 
         primary_file = validated_data.pop('primary_file', None)
         if primary_file:
-            self.set_field(preprint.set_primary_file, primary_file, auth)
-            save_node = True
+            raise exceptions.ValidationError(
```

Comment:
With Edit coming out soon, it may be possible for a user to start a preprint and decide that they uploaded the wrong file before publishing -- this would prevent them from correcting that. 

Instead, could you add a check for `is_published` in `PreprintService.set_primary_file`?